### PR TITLE
SO-5684: eliminate superfluous branch fetch request

### PR DESCRIPTION
**Pull request by cmark**
This PR changes the core branch access request flow to eliminate a superfluous, unnecessary branch fetch request to gain a small speed improvement when performing requests.

### CODE REVIEW

File: commons/com.b2international.index/src/com/b2international/index/revision/DefaultRevisionIndex.java [hunk 1]
```diff
@@ -301,7 +300,7 @@ public void removeHook(Hook hook) {
 	 * @return
 	 */
 	List<Hooks.Hook> getHooks() {
-		return ImmutableList.copyOf(hooks);
+		return List.copyOf(hooks);
```
Comment (apeteri) [inline commons/com.b2international.index/src/com/b2international/index/revision/DefaultRevisionIndex.java:303]: This conversion is fine, but there is a "fun" difference between `ImmutableSet#copyOf` and `Set#copyOf` &ndash; the latter uses hashing and iteration order is not consistent.

File: commons/com.b2international.index/src/com/b2international/index/revision/RevisionBranchRef.java [hunk 2]
```diff
@@ -18,120 +18,133 @@
...
 	}
-	
+
 	@Override
 	public String toString() {
-		return MoreObjects.toStringHelper(this)
-				.add("branchId", branchId)
-				.add("segments", segments)
-				.toString();
+		return MoreObjects.toStringHelper(this).add("branchId", branchId).add("segments", segments).toString();
```
Comment (apeteri) [inline commons/com.b2international.index/src/com/b2international/index/revision/RevisionBranchRef.java]: We could list all branches as well as the "deleted" flag here.

**Review by apeteri — APPROVED** [review]
👍

File: commons/com.b2international.index/src/com/b2international/index/revision/RevisionBranchRef.java (hunk 2 above)
Comment (cmark) [reply commons/com.b2international.index/src/com/b2international/index/revision/RevisionBranchRef.java]: Done

**Review by nagyo — APPROVED** [review]
LGTM!

**Pull request merged**